### ISSUE TITLE
[Feature][Backend] Establish a standard error logging convention

### ISSUE THREAD

**Issue by keon94** [apache/incubator-devlake#3068]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Currently the codebase is filled with random usages of logger.Error(err, "msg"). Calls to this function must be done where the given "err" is not going to be returned back to the caller. Otherwise we keep spamming the logs with the same error over and over. Another use-case is when a parent logger creates a child logger using logger.Nested(). If an error happens at the child level, the function that created the child should print the error and not pass it up to the parent logger.

### Use case

To summarize the description:
1. logger.Error(err, msg) with err != nil should be called when the err is not being returned to the caller
2. logger.Error(err, msg) with err != nil should be called at functions that create that logger. That err must not be returned back to the caller, but it's ok to create a new, non-wrapped error and pass that to the caller.

Otherwise, logger.Error with err != nil should not be called because it'll lead to confusing and spammy log messages.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by Startrekzky** [comment]
Is this one duplicated with #1447 or #2134 ?

**Comment by keon94** [comment]
> Is this one duplicated with #1447 or #2134 ?

@Startrekzky No, this is standalone. This is about where in our code we should be logging error messages - not what the error messages should be.

**Comment by Startrekzky** [comment]
What's the status of this issue? @klesh

**Comment by klesh** [comment]
> What's the status of this issue? @klesh

Not sure, I think it is not yet started.

**Comment by Startrekzky** [comment]
Is this duplicated with #4282? Shall we close one of them?  @klesh @keon94

**Comment by keon94** [comment]
@Startrekzky not really. This has to do with where in our code should we be logging the errors to the backend log files. It has nothing to do with the user side.
